### PR TITLE
fix: improving URLs on linuxbrew

### LINKED ISSUE #1893: Releasing to a Linux-only Homebrew formula
**Describe the bug**

```
$ brew readall
Error: Invalid formula: /usr/local/Homebrew/Library/Taps/muesli/homebrew-tap/deckmaster.rb
formulae require at least a URL
```

The formula is generated with GoReleaser and does not target macOS. In such case, at least `depends_on :linux` should be added, but, regardless, it does not work due to the fact that macOS requires a `url` property, which is not defined on macOS.

**To Reproduce**

https://github.com/muesli/deckmaster/blob/master/.goreleaser.yml

**Expected behavior**

The generated formula should have `depends_on :linux` and handle gracefully when the tap, which may have formulas available on macOS, too, errors out due to unset `url` property.

**Environment (please complete the following information):**

- OS: mac
- OS version: macOS 11.0.1 Big Sur
- GoReleaser Version 0.146.0

### PR DESCRIPTION
**Pull request by caarlos0**
- rename some template fields 
- changing the ifs to avoid empty if statements
- adding `depends_on :linux` if only linux URLs

closes #1893
maybe helps with #1804

### CODE REVIEW

File: internal/pipe/brew/brew.go [hunk 1]
```diff
@@ -310,21 +310,21 @@ func dataFor(ctx *context.Context, cfg config.Homebrew, cl client.Client, artifa
 			result.MacOS = down
 		} else if artifact.Goos == "linux" {
 			switch artifact.Goarch {
-			case "386", "amd64":
-				if result.Linux.DownloadURL != "" {
+			case "amd64":
```
Comment (caarlos0) [inline internal/pipe/brew/brew.go:313]: we never include i386 builds, so this part of the switch was useless

File: internal/pipe/brew/testdata/test.rb.golden [hunk 2]
```diff
@@ -4,24 +4,21 @@ class Test < Formula
   homepage "https://google.com"
   version "0.1.3"
   bottle :unneeded
-
```
Comment (radeksimko) [inline internal/pipe/brew/testdata/test.rb.golden:7]: I think that extra space was actually useful from grouping perspective.

**Review by radeksimko — APPROVED** [review]
I wonder how soon are we going to need to make a similar change to account for `darwin/arm64` (the Apple Silicon) 😄

**Comment by caarlos0** [comment]
Once go 1.16 is out I believe 😂

**Review by mgrachev — APPROVED** [review]

**Review by mavogel — APPROVED** [review]
LGTM. Fixed the `golden` files to make the tests green and docs

**Comment by caarlos0** [comment]
> LGTM. Fixed the `golden` files to make the tests green and docs

the tests were right to fail, bad sed on my side... reverted your commit...

**Pull request merged**